docs/docs/guides/testing.md

```diff
@@ -114,31 +114,71 @@ function Title() {
 ```jsx
 describe('Title Component', () => {
   test('display the title correctly', async () => {
-    const mockState = { title: 'test title' };
+    const mockState = {title: 'test title'};
     global.fetch = jest.fn(() =>
       Promise.resolve({
         json: () => Promise.resolve(mockState),
-      })
+      }),
     );
-    
+
     render(
       <RecoilRoot>
         <Suspense fallback={<div>loading...</div>}>
           <Title />
         </Suspense>
-      </RecoilRoot>
+      </RecoilRoot>,
     );
     await flushPromisesAndTimers();
-    
-    expect(screen.getByText(mockState.title)).toBeInTheDocument()
-    expect(screen.getByText('loading...')).not.toBeInTheDocument()
+
+    expect(screen.getByText(mockState.title)).toBeInTheDocument();
+    expect(screen.getByText('loading...')).not.toBeInTheDocument();
+  });
+});
+```
+
+## Testing Recoil state inside a Custom Hook
+
+Sometimes it becomes convenient to write a custom React hook that relies on Recoil states. Since you test the hook in a React context, it also needs to be wrapped in `RecoilRoot`. Luckily, you can use [React Hooks Testing Library](https://react-hooks-testing-library.com/) to achieve this.
+
+### Example: React Hooks Testing Library
+
+#### State
+
+```jsx
+const countState = atom({
+  key: 'countAtom',
+  default: 0,
+});
+```
+
+#### Hook
+
+```jsx
+const useMyCustomHook = () => {
+  const [count, setCount] = useRecoilState(countState);
+  // Insert other Recoil state here...
+  // Insert other effects here...
```

Comment:
Maybe just "Insert other hook logic here...", which may or may not be an "effect"